dojo/tools/dependency_check/parser.py

```diff
@@ -239,18 +187,18 @@ def get_finding_from_vulnerability(self, dependency, related_dependency, vulnera
             reference_detail = ''
             for reference_node in references_node.findall(namespace +
                                                           'reference'):
-                name = self.get_field_value(reference_node, 'name', namespace)
-                source = self.get_field_value(reference_node, 'source', namespace)
-                url = self.get_field_value(reference_node, 'url', namespace)
-                reference_detail += 'name: {0}\n' \
-                                     'source: {1}\n' \
-                                     'url: {2}\n\n'.format(name, source, url)
+                name = reference_node.findtext(f"{namespace}name")
```

Comment:
I have a suggestion for the references: 

```python
            for reference_node in references_node.findall(namespace + 'reference'):
                ref_source = reference_node.findtext(f"{namespace}source")
                ref_url = reference_node.findtext(f"{namespace}url")
                ref_name = reference_node.findtext(f"{namespace}name")
                if ref_url == ref_name:
                    reference_detail += f'**Source:** {ref_source}\n' \
                                        f'**URL:** {ref_url}\n\n'
                else:
                    reference_detail += f'**Source:** {ref_source}\n' \
                                        f'**URL:** {ref_url}\n' \
                                        f'**Name:** {ref_name}\n\n'
```

* There was a bug, because the variable `name` is reused here and overwrites the variable carrying the name of the vulnerability, which is later used for the title.
* Preserves the order of the 3 attributes as they are in the XML report.
* `URL` is an abbreviation and is written in capitals in English.
* In a real life example I used to test it, the `URL` and the `name` where the same a few times, so the references don't repeat the same data here anymore. Not completely sure if it may confuse people when the `name` is missing sometimes.